Cabal-QuickCheck/src/Test/QuickCheck/Instances/Cabal.hs

```diff
@@ -503,14 +503,25 @@ shortListOf1 bound gen = sized $ \n -> do
     vectorOf k gen
 
 arbitraryShortToken :: Gen String
-arbitraryShortToken = arbitraryShortStringWithout "{}[]"
+arbitraryShortToken = arbitraryShortStringWithout "{}[]" (const True)
 
 arbitraryShortPath :: Gen String
-arbitraryShortPath = arbitraryShortStringWithout "{}[],"
-
-arbitraryShortStringWithout :: String -> Gen String
-arbitraryShortStringWithout excludeChars =
-    shortListOf1 5 $ elements [c | c <- ['#' ..  '~' ], c `notElem` excludeChars ]
+arbitraryShortPath = arbitraryShortStringWithout "{}[],<>:|*?" (not . winDevice)
+    where
+        winDevice path = let p' = map toLower path
+                          in any (isDev p') ["con", "aux", "prn", "com", "lpt", "nul"]
+        isDev path dev = dev `isPrefixOf` path ||
+                         ('\\' : dev) `isInfixOf` path ||
+                         ('/' : dev) `isInfixOf` path
```

Comment:
Isn't this too crude?  It will not only exclude `con` but also `contact` etc.

Reply:
Yes, but `con.x` (for any `x`) also has to be excluded. This quickly becomes complicated.

Reply:
I redesigned it a bit; it still excludes a little more than it needs to, but being exact will be fairly expensive. Really, the right solution is to outsource it to a proper `FilePath` type. 😀

Reply:
> Really, the right solution is to outsource it to a proper `FilePath` type.

Do/Can we have this captured as an issue?

Reply:
I am adding comments about it as well.

Reply:
#9281.